_applications/my-first-radanalytics-app.adoc

```diff
@@ -0,0 +1,113 @@
+
+[[setup]]
+== Setup your coding environment
+
+To utilize the source-to-image workflow you will need to have a https://git-scm.com[Git] repository to store your work. This repository is then used by the source-to-image builders to create the final application image which will be run on OpenShift. For the purposes of this tutorial your repository will need to be accessible by the OpenShift cluster, the easiest way to do this is by using an external service like https://github.com[GitHub] or https://bitbucket.com[BitBucket].
+
+Create a new repository on your preferred platform and make a local clone of it, if you need help with this process there are instructions available in https://confluence.atlassian.com/get-started-with-bitbucket/create-a-repository-861178559.html[BitBucket's documentation] and https://help.github.com/articles/create-a-repo/[GitHub's documentation].
+
+With your freshly minted repository you are ready to start coding!
+
+[[stack]]
+== Choose your language and technology stack
+
+This tutorial has been developed for multiple languages and HTTP frameworks. Functionally, all these microservices work in the same manner with similar results to an end user. In this way they represent a black box style microservice which could be replaced with another component that satisfies the same input and output requirements.
+
+Each of these language specific tutorials provide instructions on how to structure your repository and what files to create. You will also learn how each microservice works and the commands necessary to build and launch them in OpenShift. Because the source-to-image templates that you will use are language specific, you must choose a single implementation to exist in your repository. When you are finished, return here to learn about interacting with your microservice and exposing it outside of OpenShift. Please choose one of the following language and framework options:
+
+* link:/assets/my-first-radanalytics-app/sparkpi-python-flask.html[Python with Flask]
+
+[[user]]
+== Become a user
+
+At this point you should have created a code repository for your microservice, populated the repository with source files, built an application image and launched that image on OpenShift. The final stage in this tutorial is to expose your microservice outside of OpenShift and begin interacting with it as a user would.
+
+The first step in this process is to expose a route to your microservice. OpenShift contains an edge router which will associate domain name URIs with services. By default, applications you create through source-to-image have services that can expose routes which will contain their name, the project name and a hostname for the OpenShift server. You can create these routes by using the following command:
+
+....
+oc expose svc/sparkpi
+....
+
+To see the routes available in your project, run `oc get routes`, you should see something like this:
```

Comment:
I don't know if we should make a note and maybe include a hyperlink (?) about the "xip.io" routes.  If they're using oc cluster up, then xip.io will be the default.  If they're using a full OpenShift install, it won't be the default unless they edit the master-config.yaml and set it up as a default domain.

What do you think is appropriate here?  A few words and a link, or nothing?

Reply:
this is a good question, i purposely did not dive into a discussion of xip.io. my feeling is that this doc is probably not the best place for a discussion of openshift cluster routes. the assumption is that they will be using an openshift cluster that is either run by `oc cluster up`, minishift, or admin'd by someone. given those assumptions, i just want the reader to know how to get the route and utilize it.

i'm content to leave it the way it is unless there are more objections.